src/components/visualizations/chart/highcharts/plugins/adjustTickAmount.ts

```diff
@@ -125,32 +125,56 @@ export function adjustTicks(axis: IHighchartsAxisExtend): void {
 }
 
 /**
- * Align zero of secondary axis to primary axis
+ * Get axis score that increase 1 for data having positive and negative values
  * @param axis
  */
-export function alignSecondaryAxis(axis: IHighchartsAxisExtend): void {
-    if (isPrimaryYAxis(axis)) {
-        // only handle on secondary axis
+export function getAxisScore(axis: IHighchartsAxisExtend): number {
+    let score: number = 0;
+    const { dataMin, dataMax } = axis;
+    const yAxisMin = Math.min(0, dataMin);
+    const yAxisMax = Math.max(0, dataMax);
+
+    if (yAxisMin < 0) {
+        score += 1;
+    }
+    if (yAxisMax > 0) {
+        score += 1;
+    }
+    return score;
+}
+
+/**
+ * Base on axis score which is bigger than another, will become base axis
+ * The other axis will be aligned to base axis
+ * @param axes
+ */
+function getBaseAxis(axes: IHighchartsAxisExtend[]): IHighchartsAxisExtend[] {
```

Comment:
The param `axes` is only yAxis. should change name of function and parameter to context of yAxis.
e.g. `getBaseYAxis(yaxes: IHighchartsAxisExtend[])`

Reply:
done